src/sentry/static/sentry/app/components/dropdownMenu.jsx

```diff
@@ -90,26 +96,42 @@ class DropdownMenu extends React.Component {
 
   // Opens dropdown menu
   handleOpen = e => {
-    let {onOpen, isOpen} = this.props;
+    let {onOpen, isOpen, alwaysRenderMenu} = this.props;
     let isControlled = typeof isOpen !== 'undefined';
     if (!isControlled) {
       this.setState({
         isOpen: true,
       });
     }
 
+    if (this.mouseLeaveId) {
+      window.clearTimeout(this.mouseLeaveId);
+    }
+
+    // If we always render menu (e.g. DropdownLink), then add the check click outside handlers when we open the menu
+    // instead of when the menu component mounts. Otherwise we will have many click handlers attached on initial load.
+    if (alwaysRenderMenu) {
+      document.addEventListener('click', this.checkClickOutside, true);
+    }
+
     if (typeof onOpen === 'function') {
       onOpen(e);
     }
   };
 
   // Decide whether dropdown should be closed when mouse leaves element
+  // Only for nested dropdowns
   handleMouseLeave = e => {
+    let {isNestedDropdown} = this.props;
+    if (!isNestedDropdown) return;
+
     let toElement = e.toElement || e.relatedTarget;
 
     try {
       if (this.dropdownMenu && !this.dropdownMenu.contains(toElement)) {
-        this.handleClose(e);
+        this.mouseLeaveId = window.setTimeout(() => {
+          this.handleClose(e);
+        }, MENU_CLOSE_DELAY);
       }
     } catch (err) {
       Raven.captureException(err, {
```

Comment:
why was this block try/caught? hard to debug problem?

Reply:
hmm I'm not sure